pkg/resources/secret_common.go

```diff
@@ -51,62 +55,36 @@ var secretCommonSchema = map[string]*schema.Schema{
 	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
 }
 
-type commonSecretCreate struct {
-	name     string
-	database string
-	schema   string
-	comment  *string
-}
-
-func handleSecretCreate(d *schema.ResourceData) commonSecretCreate {
-	create := commonSecretCreate{
-		name:     d.Get("name").(string),
-		database: d.Get("database").(string),
-		schema:   d.Get("schema").(string),
-	}
-	if v, ok := d.GetOk("comment"); ok {
-		create.comment = sdk.Pointer(v.(string))
+func handleSecretImport(d *schema.ResourceData) error {
+	if _, err := ImportName[sdk.SchemaObjectIdentifier](context.Background(), d, nil); err != nil {
+		return err
 	}
+	return nil
+}
 
-	return create
+func handleSecretCreate(d *schema.ResourceData) (database, schema, name string) {
+	return d.Get("database").(string), d.Get("schema").(string), d.Get("name").(string)
 }
 
 func handleSecretRead(d *schema.ResourceData, id sdk.SchemaObjectIdentifier, secret *sdk.Secret, secretDescription *sdk.SecretDetails) error {
-	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
-		return err
-	}
-	if err := d.Set("name", id.Name()); err != nil {
-		return err
-	}
-	if err := d.Set("database", secret.DatabaseName); err != nil {
-		return err
-	}
-	if err := d.Set("schema", secret.SchemaName); err != nil {
-		return err
-	}
-	if err := d.Set("comment", secret.Comment); err != nil {
-		return err
-	}
-	if err := d.Set(ShowOutputAttributeName, []map[string]any{schemas.SecretToSchema(secret)}); err != nil {
-		return err
-	}
-	if err := d.Set(DescribeOutputAttributeName, []map[string]any{schemas.SecretDescriptionToSchema(*secretDescription)}); err != nil {
+	err := errors.Join(
+		d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()),
+		d.Set("comment", secret.Comment),
+		d.Set(ShowOutputAttributeName, []map[string]any{schemas.SecretToSchema(secret)}),
+		d.Set(DescribeOutputAttributeName, []map[string]any{schemas.SecretDescriptionToSchema(*secretDescription)}),
+	)
+	if err != nil {
```

Comment:
`return errors.Join...`

Reply:
this is returning err which is of type `errors.Join()` or am I wrong and this does not work?

Reply:
errors.Join takes into consideration only non-nil errors. So if all are nil, then the result is also nil. So there is no need to do this if and two return, you can simply return errors.Join(...)

Reply:
ok thank you for clarification